server/api/products.js

```diff
@@ -8,47 +8,76 @@ router.get('/page/:offset', async (req, res, next) => {
   try {
     let offset = Number(req.params.offset)
     const products = await Product.findAll({
+      include: [{model: Photo}],
       limit: 20,
-      offset: 20*offset
+      offset: 20 * offset
     })
     res.json(products)
   } catch (err) {
     next(err)
   }
 })
 
+router.get('/search/:term/page/:offset', async (req, res, next) => {
+  try {
+    let offset = Number(req.params.offset)
+    let query = req.params.term
+    const products = await Product.findAll({
+      include: [{model: Photo}],
+      where: {
+        title: {like: '%' + query + '%'}
+      },
+      limit: 20,
+      offset: 20 * offset
+    })
+    res.json(products)
+  } catch (err) {
+    next(err)
+  }
+})
 
 router.get('/categories/:categoryId/page/:offset', async (req, res, next) => {
   try {
     let id = parseInt(req.params.categoryId, 10)
     let offset = Number(req.params.offset)
     let results = await Category.findAll({
-      include: [{model: Product}],
+      include: [{model: Product, include: [{model: Photo}]}],
       where: {
         id: id
       },
       limit: 20,
-      offset: 20*offset
+      offset: 20 * offset
     })
-    if (results[0]){
+    if (results[0]) {
       let desiredProducts = results[0].products
-      console.log('results', desiredProducts)
       res.json(desiredProducts)
-    }
-    else {
-      return []
+    } else {
+      res.json([])
     }
   } catch (err) {
     next(err)
   }
 })
 
-
 router.get('/:id', async (req, res, next) => {
   try {
     let id = req.params.id
-    const products = await Product.findOne({ where: {id} })
-    res.json(products)
+    const product = await Product.findOne({
+      where: {id},
+      include: [{model: Photo}]
+    })
+    res.json(product)
+  } catch (err) {
+    next(err)
+  }
+})
+
+//FIX GET REQUEST ABOVE
+router.post('/:id', async (req, res, next) => {
+  try {
+    console.log(req.body)
+    const review = await Reviews.create(req.body)
+    req.json(review)
   } catch (err) {
     next(err)
   }
```

Comment:
I'm confused about how these routes are different from routes for posting/getting products

Reply:
Agree.  The routes I got confused.  I worked on them a bit tonight to put reviews in there own routes with product.  Need to review all of reviews slice Tomorrow with someone and go over routes.